bridge-token-factory/src/lib.rs

```diff
@@ -378,6 +395,50 @@ impl BridgeTokenFactory {
             Balance::from(current_storage - initial_storage) * STORAGE_PRICE_PER_BYTE;
         required_deposit
     }
+
+    pub fn set_metadata(
+        &self,
+        address: String,
+        name: Option<String>,
+        symbol: Option<String>,
+        reference: Option<String>,
+        reference_hash: Option<Base64VecU8>,
+        decimals: Option<u8>,
+        icon: Option<String>,
+    ) -> Promise {
+        assert!(self.controller_or_self());
+        ext_bridge_token::set_metadata(
+            name,
+            symbol,
+            reference,
+            reference_hash,
+            decimals,
+            icon,
+            &self.get_bridge_token_account_id(address),
+            env::attached_deposit(),
+            env::prepaid_gas() - OUTER_SET_METADATA_GAS,
+        )
+    }
+
+    pub fn get_controller(&self) -> Option<AccountId> {
+        env::storage_read(CONTROLLER_STORAGE_KEY)
+            .map(|value| String::from_utf8(value).expect("Invalid controller account id"))
```

Comment:
WDYT, should we add another prefix for storage beside the `CONTROLLER_STORAGE_KEY`? Or are you considering that `a` in the `CONTROLLER_STORAGE_KEY` as that prefix?

Reply:
Yep, I'm considering the `a` as that prefix. Didn't want to over-engineer there since we don't need it. But the idea is to use `a` as a prefix for custom variables to avoid potential conflicts in the future.

Reply:
Ok, I agree!